poc/Cornelius.tbasic

```diff
@@ -0,0 +1,762 @@
+﻿'---Script created on 04-25-2020 20:39:06 by
+uses "console"
+Uses "TBGL"
+Uses "UI"
+
+#include "gameZoneDefinition.tbasicu"
+#Include "https://github.com/DirectuX/logger/raw/master/LOG.tbasicu"
+
+Global logger as tLogThis
+logger.defaultTarget = %LT_ToConsole
+Console_SetScreenBufferSize(120,300)
+
+  Type tRiver
+      minSectorSurface as dWord
+      minHalfWidth as byte
+      maxHalfWidth as byte
+      banksWidth as byte
+      seed as Number
+      meandering as single
+      meander as Single
+End Type
+
+  Type tXY
+      x as DWORD
+      y as DWORD
+End Type
+
+' To store flow direction in waypoints
+  begin Const '
+  %NORTH         = 1
+  %SOUTH         = 2
+  %EAST          = 4
+  %WEST         = 8
+  end const
+
+Function TBMain()
+
+  logger.defaultTitle = Function_Name
+  logger.wInfo("Logger loaded")
+
+  call parametersSetting ' Sets initial variables
+  Call initDisplay ' Create the windows for map display
+  Call initWorldMap ' Build straight river
+  Call stirWorldMap
+  CAll sleepDisplay ' Final loop to keep displaying map
+
+End Function
+
+function parametersSetting()
+
+  logger.defaultTitle = Function_Name
+
+  global displayWindowSize as tXY
+  global hWnd AS DWORD
+
+  global myRiver as tRiver
+  Global actualRiverStatus as tRiver
+
+  myRiver.banksWidth = 3
+  myRiver.minHalfWidth = 1
+  myRiver.maxHalfWidth = 4 ' typical = 3 to 12
+  myRiver.minSectorSurface = 100 * myRiver.maxHalfWidth
+  myRiver.meandering = 0.15 ' how is the meander distribution ( far-off and nearby disclosed % ) typical = 0.15
+  myRiver.meander = 0.9 ' how max long is meander ( is a % of stretch length ) typical = 0.9
+  myRiver.seed = 23031 ' try different values ' 23031
+
+  Global maxLarge as Byte = 1 + myRiver.maxHalfWidth * 2 ' max river width, this is also the lakes's minimum side
+  global mapDimensions as tXY
+
+  for i as byte = myRiver.minHalfWidth to myRiver.maxHalfWidth
+    mapDimensions.y += Fix(myRiver.minSectorSurface/i^0.5)
+  next i
+
+  mapDimensions.x = fix(mapDimensions.y / 4)
+
+  global actualPosition as tXY
+
+'[todo] add a calm pool there (minimum lake size should be ok)
+'Start (source) of the river
+  actualPosition.y = 0
+  actualPosition.x = fix(mapDimensions.x / 2)
+
+  global map as tGameZoneDefinition()
+
+  logger.write(%LT_ToMsgBox + %LT_ToConsole, mapDimensions.x & " x " & mapDimensions.y, "mapDimensions", %TAG_INFO)
+
+End Function
+
+function initDisplay()
+
+  DESKTOP GET CLIENT TO displayWindowSize.x, displayWindowSize.y
+  
+  MsgBox 0, str$(displayWindowSize.x) & " x " & str$(displayWindowSize.y),,"Desktop resolution"
```

Comment:
Not crucial but when Y resolution goes from 1040 to 1080 then window is no more docked to the right; it gets centred !
(Task bar hiding)